test/e2e-suite-2/functional_test.go

```diff
@@ -52,9 +52,12 @@ func (s *FunctionalSuite) TestDropOnFull() {
 
 	w.SendMessageTo(pipelineName, "in", NewHttpPostRequest().WithBody([]byte("1")))
 	// give buffer writer some time to update the isFull attribute.
-	// 5s is a carefully chosen number to create a stable buffer full scenario, it is a bit higher than the buffer writer's refresh interval.
-	time.Sleep(time.Second * 5)
+	// 10s is a carefully chosen number to create a stable buffer full scenario.
+	time.Sleep(time.Second * 10)
 	w.SendMessageTo(pipelineName, "in", NewHttpPostRequest().WithBody([]byte("2")))
+	// there could be a race condition between the buffer write and the sink scaling up, if the buffer write happens after the sink scaling up, message 2 will not be dropped.
```

Comment:
Check the metrics to verify it has n message dropped? And in this case, we might not need to do the rest?

Reply:
Thanks Derek! I didn't know there's a way to retrieve number of dropped message, is it through port-forwarding with VertexMetricsPort? And if we are able to verify n message dropped through metrics, should the rest checks guarantee to succeed?

Reply:
That's what I'm thinking, verifying the dropped messages and processed messages in the metics, and then skip the rest.